pkg/controller/init.go

```diff
@@ -882,3 +884,18 @@ func (c *Controller) syncFinalizers() error {
 	klog.Info("sync finalizers done")
 	return nil
 }
+
+func (c *Controller) ReplaceFinalizer(cachedObj client.Object) ([]byte, error) {
+	if slices.Contains(cachedObj.GetFinalizers(), util.DepreciatedFinalizerName) {
+		newObj := cachedObj.DeepCopyObject().(client.Object)
+		controllerutil.RemoveFinalizer(newObj, util.DepreciatedFinalizerName)
+		controllerutil.AddFinalizer(newObj, util.KubeOVNControllerFinalizer)
+		patch, err := util.GenerateMergePatchPayload(cachedObj, newObj)
+		if err != nil {
+			klog.Errorf("failed to generate patch payload for %s, %v", newObj.GetName(), err)
+			return nil, err
+		}
+		return patch, nil
+	}
+	return nil, nil
+}
```

Comment:
We can use `https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#Client` to simplify codes.

Reply:
i tried, but it is hard for now.